### PR TITLE
Negative strand option for liftover

### PR DESCRIPTION
**Pull request by jigold**
@konradjk @bw2 Can you double check the test examples in `test_liftover_negative_strand` and give feedback on the interface?

### CODE REVIEW

File: hail/python/hail/expr/functions.py [hunk 1]
```diff
@@ -4211,8 +4212,13 @@ def liftover(x, dest_reference_genome, min_match=0.95):
...
     dest_reference_genome : :obj:`str` or :class:`.ReferenceGenome`
         Reference genome to convert to.
-    min_match : :class:`.Expression` of type :py:data:`.tfloat64`
+    min_match : :obj:`float`
         Minimum ratio of bases that must remap.
+    include_negative_strand : :obj:`bool`
+        If True, output the result as a :class:`.StructExpression` with the first field `result` being
+        the locus or locus interval and the second field `is_negative_strand` is a boolean indicating
+        whether the locus or locus interval has been mapped to the negative strand of the destination
+        reference genome.
```
Comment (konradjk) [inline hail/python/hail/expr/functions.py]: Needs an `Otherwise, ...`

**Comment by bw2** [comment]
At first glance, the tests look fine to me. More thorough tests could be done using the data files here
https://github.com/broadinstitute/picard/blob/e0bb690d57f73fd2495fc5a77b497e9696f51f81/src/test/java/picard/util/LiftoverVcfTest.java#L65-L99

The interface also looks fine in terms of a non-breaking way to add strand info.
Is the plan to use this to implement a hl.liftover(mt) function that includes the checks from https://github.com/broadinstitute/picard/blob/master/src/main/java/picard/vcf/LiftoverVcf.java#L350-L397 ?

**Comment by jigold** [comment]
Thanks! I'll look into this later. The plan is to have a `liftover_rows` method that does the same thing as `LiftoverVCF`. However, this requires infrastructure to do a while loop, which needs to be built first.

**Comment by bw2** [comment]
oh, interesting.. I thought liftover processed each variant independently. Just out of curiosity, what's the purpose of the special while-loop infrastructure?
and will it be possible to avoid shuffling when sorting lifted-over variants?

**Comment by jigold** [comment]
The `liftover` function is just a mapping from old locus to new locus. The **method** I'm thinking of implementing will do the left aligning and strand flipping of the alleles as well. I haven't decided if we should be trying to flip the genotypes at all or some of the more complicated things that function does. The while loop is needed to left align the indels in the case the strand flipped. See this code: https://github.com/broadinstitute/picard/blob/dee16518d89838fac40c6f9c07f342430326cb16/src/main/java/picard/util/LiftoverUtils.java#L365-L428

**Comment by jigold** [comment]
I looked at the picard tests and realized they just made a fake chain file with one interval that mapped to the negative strand in the destination reference. My existing test covers this scenario -- one interval maps to the positive strand of the destination and the other to the negative strand. Since I'm only passing the strand information from Picard, I'm happy with the tests I have already.

**Comment by konradjk** [comment]
I trust that it's fine then (and if it's not, I'm sure we'll come by to complain at that point 😄). @patrick-schultz over to you

**Pull request merged**